procurement_auto_create_group/models/stock_rule.py

```diff
@@ -16,15 +16,18 @@ def _onchange_group_propagation_option(self):
         if self.group_propagation_option != "propagate":
             self.auto_create_group = False
 
+    def _get_auto_procurement_group(self, product):
+        group_data = self._prepare_auto_procurement_group_data(product)
+        return self.env["procurement.group"].create(group_data)
+
     def _push_prepare_move_copy_values(self, move_to_copy, new_date):
         new_move_vals = super()._push_prepare_move_copy_values(move_to_copy, new_date)
         if self.auto_create_group:
-            group_data = self._prepare_auto_procurement_group_data()
-            group = self.env["procurement.group"].create(group_data)
+            group = self._get_auto_procurement_group(move_to_copy.product_id)
             new_move_vals["group_id"] = group.id
         return new_move_vals
 
-    def _prepare_auto_procurement_group_data(self):
+    def _prepare_auto_procurement_group_data(self, product):
```

Comment:
Any override of this method will break once merged.
In OCA, it is only overridden in `procurement_auto_create_group_carrier` which is not yet available on 16.0 (mig PR here => https://github.com/OCA/stock-logistics-workflow/pull/1718)